corehq/apps/geospatial/static/geospatial/js/gps_capture.js

```diff
@@ -11,6 +11,37 @@ hqDefine("geospatial/js/gps_capture",[
     initialPageData
 ) {
     'use strict';
+    const selectedMarkerColor = "#00FF00"; // Green
+
+    const MAP_CONTAINER_ID = "geospatial-map";
+    var map;
+    var selectedDataListItem;
+    var coordinateCaptureMarker;
+
+    function setMapVisible(isVisible) {
+        if (isVisible) {
+            $("#" + MAP_CONTAINER_ID).show();
+            centerMapWithMarker();
+        } else {
+            $("#" + MAP_CONTAINER_ID).hide();
+        }
+    }
+
+    function centerMapWithMarker() {
+        if (selectedDataListItem) {
+            let dataItem = selectedDataListItem.itemLocationBeingCapturedOnMap();
+            if (dataItem.lon() && dataItem.lat()) {
+                map.setCenter([dataItem.lon(), dataItem.lat()]);
+                setMarkerAtLngLat(dataItem.lon(), dataItem.lat());
+            }
+        }
+    }
+
+    function runSaveSuccessCallback() {
```

Comment:
Should this function be run when the user changes pages? I'm just thinking of a scenario where the user has clicked "Capture on Map" but goes to another page before saving. In this scenario, the map will still be referencing a marker from a different page, until the user clicks the capture button for a different row. 

This certainly feels like an edge-case situation, but I'm curious to hear if you think this is something we should account for @Charl1996?

Reply:
Maybe I'll change it as you suggested, because that way it's never ambiguous to the user what's happening on the screen.

Reply:
nit: a better name could be resetMap.

Reply:
Yeah, that's a good one